lambda/translclass.ml

```diff
@@ -215,6 +215,9 @@ let rec build_object_init_0
     Tcl_let (_rec_flag, _defs, vals, cl) ->
       build_object_init_0
         ~scopes cl_table (vals@params) cl copy_env subst_env top ids
+  | Tcl_open (_descr, cl) ->
+      build_object_init_0
+        ~scopes cl_table params cl copy_env subst_env top ids
   | _ ->
```

Comment:
Having an exhaustive list of constructor instead of `_` would probably prevent future instances of the bug

Reply:
I don't agree. If we followed the specification from the manual to the letter, then only `Tcl_let` should trigger the lifting, and this PR is actually introducing a bug. I felt that the spirit of the specification should allow `Tcl_open` too, so I opened this PR, but in my mind it's clear that those two cases are exceptions. If a constructor gets added, it should fall into the catch-all, unless we specifically want to extend the toplevel optimisation to it again (and I argue in #13178 that this optimisation is not even a good idea).

Reply:
> If a constructor gets added, it should fall into the catch-all, unless we specifically want to extend the toplevel optimisation to it again

Having the compiler complain gives the opportunity to think about whether to extend the optimization or not. Having a diff here when adding a constructor will make this decision explicit to other reviewers as well.

Reply:
I agree with @hhugo that it would be better to list all other constructors explicitly here.

Reply:
I still think it's a bad idea. It makes it clear that we're not handling toplevel lets behind a constraint, or in a class application, in a consistent way.
If we have a fragile pattern-matching, this can be excused: it's just a hack, and it's actually documented as such in the manual. If we introduce exhaustive pattern-matching, it looks like a conscious decision to have an inconsistent compilation scheme, and I don't want to be involved in that.
If you want exhaustive pattern-matching, you're free to introduce it in a further commit.

Reply:
Thanks !